### DISCUSSION TITLE
How to change the java version in codespaces?

### DISCUSSION THREAD

**Discussion by lijinhong11** [community/community#47246] [Codespaces] [answered]
### Select Topic Area

Product Feedback

### Body

I want use other java version to develop on codespaces. Such as 8, 16, 17. But there is only java 11.

**Comment by KartikM123** [comment]
Hi @lijinhong11 ! Thanks for reaching out!

To use other versions of java on your codespace, I recommend configuring your devcontainer to point to a specific version. We have a great guide here on how to do that specifically for java projects! https://docs.github.com/en/codespaces/setting-up-your-project-for-codespaces/adding-a-dev-container-configuration/setting-up-your-java-project-for-codespaces. 

Let me know if that helps!

**Reply by paddunambala** [reply]
hi sir

**Reply by 29berira-lab** [reply]
Hi

**Reply by 29berira-lab** [reply]
Are you OK?

**Reply by lijinhong11** [reply]
what happen there

**Reply by 29berira-lab** [reply]
nothing it’s OK no problem. Come down.

**Reply by iranejejefrank76-arch** [reply]
Frank Tech

**Reply by bmw4268000000001-rgb** [reply]
Hello

**Comment by albertopasqualetto** [comment, marked as answer]
Use SDKMAN! (already installed in codespaces).

**Reply by mr-poston** [reply]
How do I use SDKMAN?

**Reply by slickbricks** [reply]
just use: (https://sdkman.io/usage#listversions)

```sdk install java-version```

for instance, to install java 17 corretto

```sdk install java 17.0.9-amzn```

to list all of the versions

```sdk list java```

**Reply by nicornk** [reply]
I needed to switch to Java 17 and this one-linter did the trick for me:

`sdk default java 17.0.9-ms`

**Reply by 29berira-lab** [reply]
Hi

**Reply by Ramchandra322** [reply]
> just use: (https://sdkman.io/usage#listversions)
> 
> `sdk install java-version`
> 
> for instance, to install java 17 corretto
> 
> `sdk install java 17.0.9-amzn`
> 
> to list all of the versions
> 
> `sdk list java`

How I used this and how to help me content creation

**Comment by GHomeSmartMiniSmartPlug** [comment]
To change the Java version in **[Codespaces](https://www.hewglo.com/irobot-roomba-i3-evo-3550-self-emptying-robot-vacuum-25503573.html)**, you can follow these steps:

- Open the terminal in your Codespace.
- Run the command java -version to check the current Java version.
- Use the command sudo update-alternatives --config java to see the available Java versions.
- Select the desired Java version by entering the corresponding number.
- Verify the version change by running java -version again.

Remember to restart any running Java applications for the changes to take effect.

**Reply by albertopasqualetto** [reply]
This did not work for me, sdkman did his job of installing and setting as default, instead.

**Reply by AlexAggarwal** [reply]
It didn't work for me too!

**Reply by bjconlan** [reply]
I dont think this will work because the default java path is using `/home/codespace/java/current/bin/java`, if you look at the `/home/codespace/java` folder you will see a number of java versions. If you just `rm current && ln -s 17.0.8-ms current` it will symlink the java 17.0.8-ms version (java 17)

**Reply by Sasha84-code** [reply]
https://www.dropbox.com/scl/fi/0ybovovhicva4853tgyg3/expertedge_products_20260108T140126451Z.csv?rlkey=hlkuk8jberczgjhijl1e7mxiv&st=f6u1qxaj&dl=0

**Reply by bmw4268000000001-rgb** [reply]
Good

**Comment by AYOVIBEZ** [comment]
DONT NO HOW TO DO IT

**Comment by ibrahimasylla669simbayah1-max** [comment]
Java 17

**Comment by jasonradford115-ui** [comment]
thanks

**Reply by 29berira-lab** [reply]
Hi

**Reply by bmw4268000000001-rgb** [reply]
مرحبا

**Reply by bmw4268000000001-rgb** [reply]
هلا وغلا

**Reply by bmw4268000000001-rgb** [reply]
What's tempting?

**Comment by 29berira-lab** [comment]
![IMG_1492](https://github.com/user-attachments/assets/87a7535b-3ae0-4c13-84b2-12dd1077bb86)
trust me, I am from America and Africa. I speaking English. You can see that picture. I am from Africa. I speaking English well I wanna practice English with my friend because the teacher she say reading it right now I like speaking in French I like Canada I am from America.

**Comment by 29berira-lab** [comment]
#182301

**Comment by 29berira-lab** [comment]
#181947 my name is berize

**Comment by Nagref** [comment]
Use SDKMAN!

**Reply by 29berira-lab** [reply]
I don’t have anything

**Comment by bmw4268000000001-rgb** [comment]
<img width="899" height="899" alt="20260630_073924" src="https://github.com/user-attachments/assets/00f5e0cb-e138-42ba-a12c-10fa040e217b" />
Amazing I want a very beautiful and attractive girl to have sex for long hours and a small slender

**Comment by bmw4268000000001-rgb** [comment]
I want a lot of money that never ends 
<img width="720" height="1600" alt="Screenshot_٢٠٢٦٠٧٣١-٢٠٤٠١٢" src="https://github.com/user-attachments/assets/c3e25e8d-6c7c-4f19-af2d-32b5ac4ecd56" />

**Comment by bmw4268000000001-rgb** [comment]
**[love ](url)**

**Comment by jg3364537-sudo** [comment]
> ### Seleccione el área temática
> Comentarios sobre el producto
> 
> ### Cuerpo
> Quiero usar otras versiones de Java para desarrollar en Codespaces, como la 8, la 16 o la 17. Pero solo está disponible Java 11.

**Comment by jg3364537-sudo** [comment]
{
  "image": "mcr.microsoft.com/devcontainers/universal:2",
  "features": {
    "ghcr.io/devcontainers/features/java:1": {
      "version": "21",
      "jdk": true,
      "installMaven": true,
      "installGradle": false
    }
  }
}

**Comment by jg3364537-sudo** [comment]
{
  "image": "mcr.microsoft.com/devcontainers/universal:2",
  "features": {
    "ghcr.io/devcontainers/features/node:1": {
      "version": "20"
    },
    "ghcr.io/devcontainers/features/java:1": {
      "version": "17"
    },
    "ghcr.io/devcontainers/features/android-sdk:1": {
      "version": "latest"
    }
  }
}

**Comment by jg3364537-sudo** [comment]
. En tu proyecto de Codespaces, en la raíz, crea una carpeta que se llame
.devcontainer

2. Dentro de esa carpeta vas a crear un archivo que se llame
devcontainer.json. El código para ese archivo te lo dejo abajo en la
sección de CÓDIGOS.

3. Guarda todo. Ahora vas a reconstruir el Codespace. Presiona Ctrl + Shift
+ P, escribe Rebuild Container y dale Enter. Espera 2 a 3 minutos.

4. Cuando termine, abre una terminal nueva y verifica que Java sea la 17.
Pones java -version. Si te dice 17, vas bien.

5. Ahora abre el archivo android/gradle.properties de tu proyecto. Al final
de ese archivo vas a pegar una línea que te dejo abajo para que Gradle use
Java 17.

6. Para probar Android dentro de Codespaces, entras a la carpeta android y
compilas. Si tu proyecto es de Expo, primero tienes que generar la carpeta
android con el comando de prebuild.

7. Para iOS no se puede compilar dentro de Codespaces porque necesita Xcode
de Mac. Lo que haces es mandarlo a compilar a la nube de Expo con el
comando eas. Te logeas y le das build para iOS. Te regresa el link del
archivo .ipa.

8. Si después necesitas otra versión de Java, no reconstruyes nada. Solo
cambias la versión con SDKMAN dentro de la misma terminal.

**Reply by v-coder247** [reply]
Hi @jg3364537-sudo , thanks for participating!

Unfortunately, we don’t currently have moderators for languages other than English. Until that changes, we need to ask that everyone use English here in the GitHub Community when posting to ensure discussions are adhering to our [Code of Conduct](https://github.com/community/community/blob/920a928ffe3ae4863b64a4ea26917046e11f5c1a/CODE_OF_CONDUCT.md). 

We’ll be locking this post for now, but please feel encouraged to create a new discussion in English if you still need support.

**Comment by jg3364537-sudo** [comment]
Si gustas puedo preparar el prompt o los mismos códigos

**Reply by v-coder247** [reply]
Hi @jg3364537-sudo , thanks for participating!

Unfortunately, we don’t currently have moderators for languages other than English. Until that changes, we need to ask that everyone use English here in the GitHub Community when posting to ensure discussions are adhering to our [Code of Conduct](https://github.com/community/community/blob/920a928ffe3ae4863b64a4ea26917046e11f5c1a/CODE_OF_CONDUCT.md). 

We’ll be locking this post for now, but please feel encouraged to create a new discussion in English if you still need support.
